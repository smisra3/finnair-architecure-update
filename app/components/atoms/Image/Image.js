// @flow
import type { Node } from 'react';

import styled from 'styled-components';
import styles from './Image.style';

type Props = {
  className: string,
  src: string,
  srcset?: string,
  sizes?: string,
  placeholderSrc?: string,
  alt: string,
  inheritedStyles?: string,
  ref: () => void
};

const Image = ({
  className,
  src,
  srcset,
  sizes,
  placeholderSrc,
  alt,
  inheritedStyles,
  ref,
  ...other
}: Props): Node => (
  <img
    className={className}
    src={src}
    srcSet={srcset || null}
    sizes={sizes || null}
    alt={alt}
    ref={ref}
    {...other}
    onError={placeholderSrc ? (event) => {
      // eslint-disable-next-line no-param-reassign
      event.target.src = placeholderSrc;
    } : null}
  />
);

Image.defaultProps = {
  srcset: '',
  sizes: '',
  placeholderSrc: '',
  inheritedStyles: '',
};

export default styled(Image)`
  ${styles};
`;
export { Image as ImageVanilla };
