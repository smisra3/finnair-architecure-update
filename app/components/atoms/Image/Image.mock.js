const defaultImage = {
  src: '/static/images/dummies/320.png',
  srcset: '/static/images/dummies/1280.png 1024w, /static/images/dummies/640.png 640w, /static/images/dummies/320.png 320w',
  sizes: '(min-width: 36em) 33.3vw, 100vw',
  placeholderSrc: '/static/images/image-placeholder.png',
  alt: 'Dummy Image',
};

export default defaultImage;
