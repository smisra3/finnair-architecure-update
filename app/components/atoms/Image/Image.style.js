import { css } from 'styled-components';

const imageStyles = css`
  max-width: 100%;
  height: auto;
`;

export default imageStyles;
