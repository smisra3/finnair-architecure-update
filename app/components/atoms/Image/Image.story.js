// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import defaultImage from './Image.mock';

// Import Vanilla Component to display markup and props
import { ImageVanilla } from './Image';
// Import Styled Component to showcase variations
import Image from './index';

storiesOf('Atoms', module).addWithChapters('Image', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <ImageVanilla {...defaultImage} className={`hide-default-sample`} />,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Responsive Image Variations',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <Image {...defaultImage} />,
        }
      ],
    },
  ],
});
