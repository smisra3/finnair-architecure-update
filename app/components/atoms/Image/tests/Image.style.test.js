import { shallow } from 'enzyme';

import Image from '../Image';

describe('Image Component with styles', () => {
  const sampleProps = {
    src: 'http://abc.com',
    placeholderSrc: 'http://xyz.com',
    alt: 'ABC',
  };

  test('should render correctly and match snapshots', () => {
    const image = shallow(<Image {...sampleProps} />);
    expect(image).toMatchSnapshot();
  });
});
