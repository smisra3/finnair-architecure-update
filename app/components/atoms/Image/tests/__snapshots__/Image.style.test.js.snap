// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component with styles should render correctly and match snapshots 1`] = `<Component />`;
